frontend/include/chpl/uast/Variable.h

```diff
@@ -151,6 +158,14 @@ class Variable final : public VarLikeDecl {
     Returns true if this Variable represents a field.
   */
   bool isField() const { return this->isField_; }
+
+  /**
+    Returns the destination expression, like 'loc' from 'on loc var x = 1'.
+  */
+  const AstNode* destination() const {
```

Comment:
This feels a bit pedantic of me, but I think we should use a word other than "destination". To me, "destination" suggests that there's somewhere "to go", when really the variable is already there. I'd prefer something like ``location`` or ``home`` or ``homeLocale``.

Reply:
I chose `destination` here to match the `destination` field on `On` AST nodes.

https://github.com/chapel-lang/chapel/blob/b7ce56f41413c7e1d04af1bb88016e5d7f7d35e9/frontend/include/chpl/uast/On.h#L84-L92

I thought consistency might be good since syntactically the `on ... var` and `on ... do` look very similar. If you find that unconvincing, I can rename this.